config/config.go

```diff
@@ -134,30 +141,39 @@ func (c *Config) Read(file string) error {
 	defer f.Close()
 
 	d := json.NewDecoder(f)
-	err = d.Decode(&c)
-	if err != nil {
-		return err
+	return d.Decode(&c)
+}
+
+// IsAuthenticated returns true if the config contains an API key.
+// This does not check whether or not that key is valid.
+func (c *Config) IsAuthenticated() bool {
+	return c.APIKey != ""
+}
+
+// homeDir caches the lookup of the user's home directory.
+func (c *Config) homeDir() (string, error) {
+	if c.home != "" {
+		return c.home, nil
 	}
-	c.configure()
-	return nil
+	return Home()
 }
 
-// Write() saves the config as JSON.
-func (c *Config) Write() error {
-	// truncates existing file if it exists
-	f, err := os.Create(c.File)
+func (c *Config) resolvePath(argPath, envPath string) (string, error) {
+	path := argPath
+	if path == "" {
+		path = envPath
+	}
+	if path == "" {
+		path = filepath.Join("~", File)
+	}
+	h, err := c.homeDir()
 	if err != nil {
-		return err
+		return "", err
 	}
-	defer f.Close()
-
-	e := json.NewEncoder(f)
-	return e.Encode(c)
+	return strings.Replace(path, "~", h, 1), nil
 }
 
-func (c *Config) configure() (*Config, error) {
-	c.sanitize()
-
+func (c *Config) setDefaults() error {
```

Comment:
Yeah, everything seemed a bit tangled before. Or perhaps: both tangled and fragmented. This time around I've kept larger chunks. I think fewer abstractions will be easier to manage.